lib/algebra.ts

```diff
@@ -83,6 +83,17 @@ export interface BaseOperation
     metadata?: Record<string, unknown>;
     type: types;
 }
+// The interface for paths is added with types corresponding to the definitions in sparql-nrt
+export interface Paths extends BaseOperation
+{
+    type : types.PATHS;
+    shortest?: boolean;
+    cyclic?: boolean;
+    start?: rdfjs.Variable | IriTerm;
+    via?: rdfjs.Variable  | IriTerm;
+    end?: rdfjs.Variable  | IriTerm;
```

Comment:
`start`, `end`, and `via` should not be optional, according to the [grammar](https://docs.stardog.com/query-stardog/path-queries).
Keep `start` and `end` as required type `rdfjs.Variable`, and add two more properties like:

```
startPattern?: IriTerm | Bgp
endPattern?: IriTerm | Bgp
```

I'm not sure if `Bgp` is the best match for `<GRAPH PATTERN>` from the grammar. Maybe check to see what the parser returns for `WHERE` clause in `SELECT`.

`via` should have type `rdfjs.Variable | PropertyPathSymbol | Bgp` (same uncertainty about `Bgp`).

Reply:
Or maybe the graph pattern is just `Operation`.